app/services/CategorisationService.scala

```diff
@@ -39,122 +39,87 @@ class CategorisationService @Inject() (
 
   def requireCategorisation(request: DataRequest[_], recordId: String)(implicit
     hc: HeaderCarrier
-  ): Future[UserAnswers] = {
-
-    val recordCategorisations =
-      request.userAnswers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
-
-    val originalCommodityCodeOpt =
-      recordCategorisations.records.get(recordId).flatMap(_.originalCommodityCode)
-
-    recordCategorisations.records.get(recordId) match {
-      case Some(_) =>
-        Future.successful(request.userAnswers)
-      case None    =>
-        for {
-          getGoodsRecordResponse           <- goodsRecordsConnector.getRecord(eori = request.eori, recordId = recordId)
-          goodsNomenclature                <- ottConnector.getCategorisationInfo(
-                                                getGoodsRecordResponse.comcode,
-                                                request.eori,
-                                                request.affinityGroup,
-                                                Some(recordId),
-                                                getGoodsRecordResponse.countryOfOrigin,
-                                                LocalDate.now() //TODO where does DateOfTrade come from??
-                                              )
-          originalCommodityCode             = originalCommodityCodeOpt.getOrElse(getGoodsRecordResponse.comcode)
-          categorisationInfo               <- CategorisationInfo.build(goodsNomenclature, Some(originalCommodityCode)) match {
-                                                case Some(categorisationInfo) => Future.successful(categorisationInfo)
-                                                case _                        => Future.failed(new RuntimeException("Could not build categorisation info"))
-                                              }
-          updatedAnswers                   <-
-            Future.fromTry(
-              request.userAnswers.set(
-                RecordCategorisationsQuery,
-                recordCategorisations.copy(records = recordCategorisations.records + (recordId -> categorisationInfo))
-              )
-            )
-          updatedAnswersCleanUpAssessments <-
-            Future.fromTry(cleanupOldAssessmentAnswers(updatedAnswers, recordId))
-          _                                <- sessionRepository.set(updatedAnswersCleanUpAssessments)
-        } yield updatedAnswersCleanUpAssessments
-    }
-  }
+  ): Future[UserAnswers] =
+    (for {
+      getGoodsRecordResponse <- goodsRecordsConnector.getRecord(eori = request.eori, recordId = recordId)
+      recordCategorisations   =
+        request.userAnswers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
```

Comment:
default here, defeats the point of having options

Reply:
Refactored